src/v/pandaproxy/schema_registry/seq_writer.cc

```diff
@@ -0,0 +1,129 @@
+//// Copyright 2021 Vectorized, Inc.
+////
+//// Use of this software is governed by the Business Source License
+//// included in the file licenses/BSL.md
+////
+//// As of the Change Date specified in that file, in accordance with
+//// the Business Source License, use of this software will be governed
+//// by the Apache License, Version 2.
+
+#include "pandaproxy/schema_registry/seq_writer.h"
+
+#include "pandaproxy/error.h"
+#include "pandaproxy/logger.h"
+#include "pandaproxy/schema_registry/client_fetch_batch_reader.h"
+#include "pandaproxy/schema_registry/storage.h"
+#include "random/simple_time_jitter.h"
+#include "vlog.h"
+
+#include <seastar/core/coroutine.hh>
+
+using namespace std::chrono_literals;
+
+namespace pandaproxy::schema_registry {
+
+// TODO: hook consume_to_store into advance_offset
+
+/// Call this before reading from the store, if servicing
+/// a REST API endpoint that requires global knowledge of latest
+/// data (i.e. any listings)
+ss::future<> seq_writer::read_sync() {
+    auto offsets = co_await _client.local().list_offsets(
+      model::schema_registry_internal_tp);
+
+    const auto& topics = offsets.data.topics;
+    if (topics.size() != 1 || topics.front().partitions.size() != 1) {
+        auto ec = kafka::error_code::unknown_topic_or_partition;
+        throw kafka::exception(ec, make_error_code(ec).message());
+    }
+
+    const auto& partition = topics.front().partitions.front();
+    if (partition.error_code != kafka::error_code::none) {
+        auto ec = partition.error_code;
+        throw kafka::exception(ec, make_error_code(ec).message());
+    }
+
+    co_await wait_for(partition.offset - model::offset{1});
```

Comment:
Why do we wait for the previous offset, and not the one that was produced?

Reply:
partition.offset is the next offset in the partition that will be written to -- we want to wait for the last offset that was written (_loaded_offset tracks the last offset seen, rather than the next offset for writes).

e.g. initially, partition.offset is zero (no messages), but we don't want to wait for the 0th message to show up in that case.  In the initial case _loaded_offset is -1, and we would `wait_for(-1)` (i.e. be ready immediately)